### PR TITLE
[docs/revamp] proof-of-concept for synchronizing API docs with Docusaurus

### PR DESCRIPTION
**Pull request by cmpadden**
## Summary & Motivation

Sphinx supports [builders](https://www.sphinx-doc.org/en/master/usage/builders/index.html) to create outputs in various formats: html, latex, etc.  However, a third party extension `sphinx_markdown_builder` is required to generate markdown files for your Python project.

This proof-of-concept generates markdown files, and then copies them to the `docs/sphinx` directory of our Docusaurus project. Here is an example of how it is rendered:

image

*Note* that Docusaurus uses an MDX parser by default, which causes problems with the markdown files output by Sphinx. For example `<` and `>` would be interpreted as HTML tags. Docusaurus _also_ supports the CommonMark format and can automatically detect if the file is mdx or md by setting `config.markdown.format` to `detect` ([reference](https://docusaurus.io/docs/markdown-features)).

```
  markdown: {
    mermaid: true,
    format: 'detect',
  },
```

_Whether this is an ideal way to synchronize our docs is tbd..._

## How I Tested These Changes

```
cd sphinx
make sync-docusaurus

cd ../docs-next
pnpm start
```

### CODE REVIEW

File: docs/docs-next/docs/sphinx/sections/api/apidocs/definitions.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  [`@repository`](repositories.md#dagster.repository) and use
+  [`with_resources`](resources.md#dagster.with_resources) as before.
+* The resources dictionary takes raw Python objects, not just instances
+  of [`ResourceDefinition`](resources.md#dagster.ResourceDefinition). If that raw object inherits from
+  [`IOManager`](io-managers.md#dagster.IOManager), it gets coerced to an [`IOManagerDefinition`](io-managers.md#dagster.IOManagerDefinition).
+  Any other object is coerced to a [`ResourceDefinition`](resources.md#dagster.ResourceDefinition).
+
+#### asset_checks
+
+Alias for field number 7
```
Comment (PedramNavid) [inline docs/docs-next/docs/sphinx/sections/api/apidocs/definitions.md:80]: What's this?

**Review by PedramNavid — COMMENTED** [review]
I don't think we should commit the converted docs, this should be part of the build process otherwise we'll end up with merge conflicts down the line. 

The sphinx extension looks fairly straightforward, and I bet we can override it as well if we need to in order to enable other components. Could be interesting.

File: docs/docs-next/docs/sphinx/sections/api/apidocs/definitions.md (hunk 1 above)
Comment (cmpadden) [reply docs/docs-next/docs/sphinx/sections/api/apidocs/definitions.md]: Weird! I'm not sure.

**Comment by cmpadden** [comment]
> I don't think we should commit the converted docs, this should be part of the build process otherwise we'll end up with merge conflicts down the line.
> 
> The sphinx extension looks fairly straightforward, and I bet we can override it as well if we need to in order to enable other components. Could be interesting.

Removed the generated files.

If we decide to, we could still have it built locally so that vale can run, and then gitignore it. This way there is a tighter feedback loop than build -> ci -> vale fails on generated docs.

**Pull request closed without merging**